File: core/src/main/scala/misc/ctci/CountCharOccurences.scala
package misc.ctci

object CountCharOccurences extends App {
  val str = "aaaccbbdcccccfffa"

  val countArray = new Array[Int](255)
  val strLen = str.size

//  for (i <- 0 until strLen)
//    countArray(str.charAt(i)) += 1
  str.foreach{ ch =>
    //println(s"ch: $ch")
    countArray(ch) += 1
  }
  println(s"count[a] ${countArray(97)}")

  var list: List[String] = Nil
  val chars = new Array[Char](strLen)
  for (i <- 0 until strLen) {
    chars(i) = str.charAt(i)
    var find = 0
    for (j  <- 0 until i)
      if (str.charAt(i) == chars(j)) {
        //println("i: " + i + "-" + str.charAt(i) + ", j:" + j  + "-" + chars(j))
        find += 1
      }

    if (find == 1) {
      list = list :+ s"${str.charAt(i)}${countArray(str.charAt(i))}"
      println(s"Number of Occurrence of ${str.charAt(i)}"
        + s" letter is:  ${countArray(str.charAt(i))}")
    }
  }
  print(s"result: ${list.mkString("")}")
}

File: core/src/main/java/misc/CountCharOccurences.java
package misc;

public class CountCharOccurences {
    public static void main(String[] args) {
        String str = "aaaccbbdcccccfffa";

        int[] count = new int[255];

        int length = str.length();

        for (int i = 0; i < length; i++) {
            count[str.charAt(i)]++;
            //System.out.println(str.charAt(i) + ": " + count[str.charAt(i)]);
        }
        System.out.println("count['a']: " + count[97]);

        char[] ch = new char[str.length()];
        for (int i = 0; i < length; i++) {
            ch[i] = str.charAt(i);
            int find = 0;
            for (int j = 0; j <= i; j++) {
                if (str.charAt(i) == ch[j]) {
                    find++;
                    //System.out.println("i: " + i + "-" + str.charAt(i) + ", j:" + j  + "-" + ch[j]);
                }
            }

            if (find == 1) {
                System.out.println("Number of Occurrence of " + str.charAt(i)
                        + " letter is: " + count[str.charAt(i)]);
            }
        }
    }
}